### ISSUE TITLE
Help pages not available !!

### ISSUE THREAD

**Issue by ferrigno** [oda-hub/frontend-astrooda#78]
The help pages are not displayed on the front end !

**Comment by burnout87** [comment]
Just checked, seems like all the images of the help are not available.

**Comment by volodymyrss** [comment]
maybe it's an older version? Look on the antares help.

**Comment by volodymyrss** [comment]
I updated from git. I must have reverted it accidentially

**Comment by volodymyrss** [comment]
but after I reverted the DB, the default values went to the old ones, reversing https://github.com/oda-hub/frontend-astrooda/issues/75

after reinstalling the module, the help pages disappeared.

so it is clear that defaults in DB are different from that in the module

**Comment by burnout87** [comment]
> 
>     * what is the impact of values in astrooda.install if what is used is in DB? where is the source of the values we should deliver, DB or module code?

As far as I know, and have always done so far, the various default values are solely set within the `.install` files.

> since @burnout87 took care of the defaults, could you please insert them into DB and make sure it works?

So, I have never inserted those directly within the DB, nor we have done it with @motame .

**Comment by volodymyrss** [comment]
> > * what is the impact of values in astrooda.install if what is used is in DB? where is the source of the values we should deliver, DB or module code?
> 
> As far as I know, and have always done so far, the various default values are solely set within the `.install` files.
> 
> > since @burnout87 took care of the defaults, could you please insert them into DB and make sure it works?
> 
> So, I have never inserted those directly within the DB, nor we have done it with @motame .

you have inserted them by running `.install` files.
So you can see the difference in the DB after the install, and provide updated DB (possibly in coordination with @motame ).

*alternatively:*

is it feasible to determine why running the install rewrites help pages?

**Comment by motame** [comment]
Fixed !
The help pages are now a part of the core module or instrument module.
The .install of the main module (astrooda) is creating a dedicated content type which can be a book page.
All necessary fields of the new content type are created with their instances beforehand.
A full description should follow in a Wiki page.
In summary, the help pages including their images are installed with the module (main or instrument) and removed at un-installation. The solution is also covering the full lifecycle of help pages.

**Comment by motame** [comment]
@ferrigno, could you please have a look in dev instance:
https://cdcidev.mtmco.net/mmoda

**Comment by volodymyrss** [comment]
it's installed, modules reinstalled, cc'd,  but it does not quite look right: https://www.astro.unige.ch/mmoda/

**Comment by motame** [comment]
With the current prod database, the modules need to be re-installed:
Disable and uninstall :
`$ drush dis -y astrooda`
(disabling the core module will disable all sub-modules, instrument modules - dependency resolution)
`$ drush pmu -y astrooda_antares astrooda_isgri astrooda_jemx astrooda_polar astrooda_spi_acs`
Delete manually the help pages and the corresponding content type ...
Then install (enable):
`$ drush en -y astrooda_antares astrooda_isgri astrooda_jemx astrooda_polar astrooda_spi_acs`
(enabling one of the instrument modules will enable automatically the core module - dependency resolution)

**Or mush much better !**
I pushed the four components in gitlab reference:
`astrooda`
`bootstrap_astrooda`
`drupal7-db-for-astrooda`
`drupal7-for-astrooda`

**Note** that the database (`drupal7-db-for-astrooda`) tables 'users' and 'users_roles' are empty !
They need to be restored from the current prod DB.

It's better to proceed with the later procedure as new modules are installed (`drupal7-for-astrooda`) and their config is in the DB (`drupal7-db-for-astrooda`).

**Comment by volodymyrss** [comment]
ok, so one problem was simple: it's because repositories were not synced from gitlab to github, because due github policy change the sync broke. I restored it, and now it will be all automatic again.

I did the reinstall as you said, I slightly modified it now to be exactly as you suggest:
https://github.com/oda-hub/frontend-chart/blob/master/make.sh#L110-L131

It's a lot better now.

There is one problem, images are not visible. They are referencing to something like:

https://www.astro.unige.ch/mmoda/sites/default/files/styles/large/public/mmoda/help/mmoda-spi-acs/processingdone.png

interestingly, if I click on the not shown image, it opens shown image with this url:

https://www.astro.unige.ch/mmoda/sites/default/files/mmoda/help/mmoda-spi-acs/processingdone.png

**Comment by motame** [comment]
I suspect that the directory site/defaut/files is not writable for the web server user.

I tried to reinstall all the modules from the web interface but it failed.

Would you please do the following in drupal instance dir :
```
$ drush dis -y astrooda
$ drush pmu -y astrooda_antares astrooda_isgri astrooda_jemx astrooda_polar astrooda_spi_acs
$ drush pmu -y astrooda
$ drush en -y astrooda_antares astrooda_isgri astrooda_jemx astrooda_polar astrooda_spi_acs

```

**Comment by motame** [comment]
I fully agree with you !
I suspect the famous ?itok=Qwkoxiq4 at the end of the URL of images to protect access to images ...
I am investigating more ...

**Comment by volodymyrss** [comment]
I did this, the `drush-reinstall-all`:

https://github.com/oda-hub/frontend-chart/blob/694b9d6f9bec856372fe64847989d0be240027c4/make.sh#L110-L128

Reinstalling shows a bunch of positive messages like:

```
Help file                                                            [status]
sites/all/modules/astrooda/instruments/astrooda_isgri/help_book/extract-isgri-spectrum-without-catalog.php
imported successfully
```

this is world-writable: `astrooda/sites/default/files/`

**Comment by motame** [comment]
The problem is related to images generated by Drupal under sites/default/files/styles.
It's quite strange.
On my dev instance, I delete completely that directory.
And one I visit any of the help pages, the corresponding images are generated on the fly ...

**Comment by volodymyrss** [comment]
> I fully agree with you !
> I suspect the famous ?itok=Qwkoxiq4 at the end of the URL of images to protect access to images ...

I find that on dev instance it works just fine without itok:

`https://cdcidev.mtmco.net/mmoda/sites/default/files/styles/large/public/mmoda/help/mmoda-isgri/topoda1.png`

> I am investigating more ...

**Comment by motame** [comment]
Are the derived images generated properly under styles directory ?
```
meharga@mtmisdcvm3:~/eclipse-workspaces/cdci/drupal7-for-astrooda$ ls -l sites/default/files/styles/
total 8
drwxr-xr-x 3 www-data www-data 4096 Sep 27 17:22 large
drwxr-xr-x 3 www-data www-data 4096 Sep 27 17:22 thumbnail
meharga@mtmisdcvm3:~/eclipse-workspaces/cdci/drupal7-for-astrooda$
```

**Comment by volodymyrss** [comment]
No, ` sites/default/files/styles/` is not created at all. I also tried to just create it with 777 rights but it does not help. 

the `sites/default/files/mmoda` directory is created during install. If I delete a file from there it returns 404, not 403. So something is not going right on the generation. Is there a log somewhere? Server log says nothing.

**Comment by motame** [comment]
https://www.astro.unige.ch/mmoda/admin/reports/dblog

**Comment by volodymyrss** [comment]
ok I can see the path it can not create but I do not see the full path. Maybe it's wrong full path but I doubt it somehow.

The `sites/default/files/` is definitely writable, and some files and directories are created there by the server, like `ctools`. If I delete them they are recreated

**Comment by motame** [comment]
I reinstalled isgri module and the derived images seems to be created ! but still access denied

**Comment by motame** [comment]
Please try the following workaround :
Add at the end of settings.php:
$conf['image_allow_insecure_derivatives'] = TRUE;

I don't like it but in between ....

**Comment by volodymyrss** [comment]
that does help, thanks!
But in dev instance, it works even without `itok`. Are you using this option too?

**Comment by volodymyrss** [comment]
at least one image is still not produced: 

https://www.astro.unige.ch/mmoda/help/mmoda

![image](https://user-images.githubusercontent.com/3909535/134950761-0f75d36b-b128-443f-b197-b7d2a94a424d.png)

all the others I checked so far seem good.

**Comment by motame** [comment]
Yeah ! it's really strange!
I'm not using that option.

**Comment by motame** [comment]
> 
> 
> at least one image is still not produced:
> 
> https://www.astro.unige.ch/mmoda/help/mmoda
> 
> ![image](https://user-images.githubusercontent.com/3909535/134950761-0f75d36b-b128-443f-b197-b7d2a94a424d.png)
> 
> all the others I checked so far seem good.

Fixed !
I'll investigate more for this one ...

**Comment by volodymyrss** [comment]
ok, thanks!
